tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -0,0 +1,197 @@
+	require.NoError(t, err, "failed to get cloud provider information")
+	resourceAuthorizer, err := auth.NewAuthorizerFromFileWithResource(azure.PublicCloud.ResourceManagerEndpoint)
+	require.NoError(t, err, "failed to get azure authorization token")
+	getFileSettings, err := auth.GetSettingsFromFile()
+	require.NoError(t, err, "failed to get info from AZURE_AUTH_LOCATION")
+	subscriptionId := getFileSettings.GetSubscriptionID()
+	vmClient := compute.NewVirtualMachinesClient(subscriptionId)
+	vmClient.Authorizer = resourceAuthorizer
+	nsgClient := network.NewSecurityGroupsClient(subscriptionId)
+	nsgClient.Authorizer = resourceAuthorizer
+	azureInfo.resourceGroupName = provider.Azure.ResourceGroupName
+	azureInfo.vmClient = vmClient
+	azureInfo.nsgClient = nsgClient
+}
+
+// readInstallerInfo reads the instanceIDs and secGroupsIDs from the
+// windows-node-installer.json file specified in "dir".
+func readInstallerInfo(t *testing.T) {
+	filePath := dir + "/windows-node-installer.json"
+	installerInfo, err := resource.ReadInstallerInfo(filePath)
+	require.NoError(t, err, "failed to read from ARTIFACT_DIR")
+	secGroupsIDs = installerInfo.SecurityGroupIDs
+	instanceIDs = installerInfo.InstanceIDs
+}
+
+// checkForWinRmPort returns a bool response if winRmHttps port is opened on
+// windows instance or not.
+func checkForWinRmPort(t *testing.T, secGroupRules []network.SecurityRule) bool {
+	var secRulePropFormat network.SecurityRulePropertiesFormat
+	var destPortRange string
+	var access network.SecurityRuleAccess
+	var protocol network.SecurityRuleProtocol
+	var priority int32
+	var sourceAddressPrefix string
+	for _, secGroupRule := range secGroupRules {
+		if checkForNil(secGroupRule.SecurityRulePropertiesFormat) {
+			continue
+		}
+		secRulePropFormat = *(secGroupRule.SecurityRulePropertiesFormat)
+		if checkForNil(secRulePropFormat.DestinationPortRange) && checkForNil(secRulePropFormat.Priority) &&
+			checkForNil(secRulePropFormat.SourceAddressPrefix) {
+			continue
+		}
+		destPortRange = *(secRulePropFormat.DestinationPortRange)
+		protocol = secRulePropFormat.Protocol
+		access = secRulePropFormat.Access
+		priority = *(secRulePropFormat.Priority)
+		sourceAddressPrefix = *(secRulePropFormat.SourceAddressPrefix)
+		if destPortRange != winRmPortHttps && access != winRmAction && protocol != winRmProtocol &&
+			priority != winRmPriority && len(sourceAddressPrefix) == 0 {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
+// testWinRmPort checks if winRmHttps port is mentioned in the inbound security
+// group rules in the worker subnet.
+func testWinRmPort(t *testing.T) {
+	ctx := context.Background()
+	var secGroupPropFormat network.SecurityGroupPropertiesFormat
+	var secGroupRules []network.SecurityRule
+	for _, nsgName := range secGroupsIDs {
+		secGroupProfile, err := azureInfo.nsgClient.Get(ctx, azureInfo.resourceGroupName, nsgName, "")
+
+		assert.NoError(t, err, "failed to get the network security group profile")
```

Comment:
Consider moving this up a line, and having the empty line come after it. The whitespace between the function call and the assert logically seperates them.